drivers/ethernet/eth_nxp_enet_qos/eth_nxp_enet_qos_mac.c

```diff
@@ -0,0 +1,622 @@
+/*
+ * Copyright 2024 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nxp_enet_qos_mac
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(eth_nxp_enet_qos_mac, CONFIG_ETHERNET_LOG_LEVEL);
+
+#include <zephyr/net/phy.h>
+#include <zephyr/kernel/thread_stack.h>
+#include <zephyr/sys_clock.h>
+#include <ethernet/eth_stats.h>
+#include "../eth.h"
+#include "nxp_enet_qos_priv.h"
+
+static const uint32_t rx_desc_refresh_flags =
+	OWN_FLAG | RX_INTERRUPT_ON_COMPLETE_FLAG | BUF1_ADDR_VALID_FLAG;
+
+K_THREAD_STACK_DEFINE(enet_qos_rx_stack, CONFIG_ETH_NXP_ENET_QOS_RX_THREAD_STACK_SIZE);
+static struct k_work_q rx_work_queue;
+
+static int rx_queue_init(void)
+{
+	struct k_work_queue_config cfg = {.name = "ENETQOS_RX"};
+
+	k_work_queue_init(&rx_work_queue);
+	k_work_queue_start(&rx_work_queue, enet_qos_rx_stack,
+			   K_THREAD_STACK_SIZEOF(enet_qos_rx_stack),
+			   K_PRIO_COOP(CONFIG_ETH_NXP_ENET_QOS_RX_THREAD_PRIORITY),
+			   &cfg);
+
+	return 0;
+}
+
+SYS_INIT(rx_queue_init, POST_KERNEL, 0);
```

Comment:
I'm sure there's a reason, but why are we setting up the RX queue in a separate init function from the rest of the ENET QOS setup?

Reply:
Makes sense- for any other reviewers, the thing I missed here is that the RX work queue is shared between all driver instances, so it needs to be initialized separately